### DISCUSSION TITLE
Should we release 0.71.4?

### DISCUSSION THREAD

**Discussion by kelset** [reactwg/react-native-releases#58] [Patches]
## Released, please refer to #59 

---

Current Release: [0.71.3](https://github.com/facebook/react-native/releases/tag/v0.71.3)

Conversations on this thread are limited:

- [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking).
- [qualified cherry-pick requests](https://reactnative.dev/contributing/release-faq#what-is-a-qualified-pick-request) of commits on main that [did not make the previous patch version](https://reactnative.dev/contributing/release-faq#how-do-i-know-if-my-fixfeature-is-in-a-certain-release).

Please include a link to the specific commit on main to be cherry-picked, for example: [facebook/react-native@bd2b7d6](https://github.com/facebook/react-native/commit/20b0eba581a00e5e7e300f6377379b836617c147)

In other words, if you cannot point to a particular commit on main, then your request likely belongs as a new issue.
If the issue is a [major release issues](https://reactnative.dev/contributing/release-faq#what-is-release-blocking), please reference the issue here.

---

#### List of qualified picks

(from oldest to newest)

1. https://github.com/facebook/react-native/commit/8b1f6e09c1b11e07096156d8d01aaff649745fbc
2. https://github.com/facebook/react-native/commit/23607aea688c950fe0cf424b280a2339636130fe
4. https://github.com/facebook/react-native/commit/3418f65d88fa8be3473f757e932ecba1bbfce783
5. https://github.com/facebook/react-native/commit/0487108461010154cf959bb0bf6ba9e82fdcc4d1
6. https://github.com/facebook/react-native/commit/4a07ed2c0c78fd7d3353aff6e0a7f75a2a4b6b86
7. https://github.com/facebook/react-native/commit/6dde1dc7cb487271567aaa9c7e093bb76c0689ad
8. https://github.com/facebook/react-native/commit/f8d8764e8f236e8495e7e5747bfe95162f3a165a
9. https://github.com/facebook/react-native/commit/c03de97fb44f0aecbec8f930bd99fe26f37f9648
10. https://github.com/facebook/react-native/commit/62a0640e4a8297177e857530e46010e83315e70a (🙅‍♂️ pick conflict)
11. https://github.com/facebook/react-native/pull/36251
12. https://github.com/facebook/react-native/pull/36323
13. (Hermes bump, see below)

Hermes side: (ref [GH action](https://github.com/facebook/hermes/actions/workflows/create-tag.yml) and [docs](https://reactnative.dev/contributing/release-branch-cut-and-rc0#1-creating-a-new-release-branch))
1. https://github.com/facebook/hermes/commit/221ce21a209e2e32a3eaaa2d9e28ca81842fad20
2. https://github.com/facebook/hermes/commit/059d08582de630af4c7b0be9925353296afbd96f
3. ~https://github.com/facebook/hermes/pull/925~
    * replaced by both https://github.com/facebook/hermes/commit/03f2dffc1d0ef8b2360a6790ad425ce4013e4de3 and https://github.com/facebook/hermes/commit/1b759f40bd2f6bb72b2a353f0d9856fcbdbb981c (see [this comment](https://github.com/facebook/hermes/issues/135#issuecomment-1448503234) for details) 	

#### Local commits to backport to main

1.

**Comment by kelset** [comment]
https://github.com/facebook/react-native/commit/28d8f25c7670d72fcac4f75dde6dd0bc9292cc03

**Comment by bigcupcoffee** [comment]
A TS fix https://github.com/facebook/react-native/commit/23607aea688c950fe0cf424b280a2339636130fe

**Comment by cortinico** [comment]
Not extremely crucial, but this enables users on 0.71+ to build from source the Android artifacts if needed, instead of pulling those from Maven Central:

- https://github.com/facebook/react-native/pull/36165
- https://github.com/facebook/react-native/pull/36128
- https://github.com/facebook/react-native/commit/f8d8764e8f236e8495e7e5747bfe95162f3a165a

That's more of an edge case use case, but could be useful for someone

**Reply by tido64** [reply]
How does this affect 0.71? The `packages/` folder would only exist if you use `facebook/react-native#main` (or similar) in `package.json`, right? Do we have a use case for `facebook/react-native#0.71-stable`?

**Reply by cortinico** [reply]
Practically that commit works in conjuction with this https://github.com/facebook/react-native-website/pull/3570
As of today, you can't trigger a build of `:ReactAndroid` from within a app project as you can't properly include it with the instructions linked in https://github.com/facebook/react-native-website/pull/3570.

We had folks in the past failing to follow that instructions (see those issues https://github.com/facebook/react-native/issues/34155#issuecomment-1199776288 https://github.com/facebook/react-native/issues/35593 as well as the Samsung ANR issues where people didn't really know how to build from source + patch package https://github.com/facebook/react-native/issues/35936#issuecomment-1404138005)

**Reply by tido64** [reply]
That makes sense. Thanks for explaining 😊

**Reply by tido64** [reply]
Should this then be backported all the way down to 0.69? You've changed the instructions and people may not have upgraded to 0.71 yet.

**Reply by cortinico** [reply]
So sadly the "build from source" page is not versioned, and I believe it's fine as it is (as I consider it an "unsupported" build path). We don't test it and we don't encourage users moving over to that build path. It should be something only for advanced users as the build is more likely to fail due to the complexity of our build logic.

For 0.71+ this is more crucial as we have artifacts on Maven Central instead of the `/android` folder, so it makes harder for users to apply patches.

If we wish, we can backport my commit down to 0.69, that's not a major change. It just makes it easier to `includeBuild(...)` react native and trigger the build from source.

**Reply by cortinico** [reply]
EDIT: I've added another commit to cherry-pick (I've edited my original message) as I got a report of a broken build from source on nightlies which I wanted to fix: https://github.com/facebook/react-native/commit/f8d8764e8f236e8495e7e5747bfe95162f3a165a

**Reply by cortinico** [reply]
EDIT: while pickign those with @kelset I've realized we missed a commit which is necessary as well (https://github.com/facebook/react-native/pull/36128). I've included it in my original comment.

**Comment by kelset** [comment]
https://github.com/facebook/react-native/commit/3418f65d88fa8be3473f757e932ecba1bbfce783

**Comment by cortinico** [comment]
This fixes a bug for relative paths with `ENTRY_FILE`: 
- https://github.com/facebook/react-native/pull/36193

(And requires a bump of RNGP)

**Comment by Kudo** [comment]
a customer reported a slow hermes bundle issue on android, which is similar to [this issue](https://github.com/facebook/hermes/issues/852#issuecomment-1352290104). do you think whether it is feasible to land [the fix](https://github.com/facebook/hermes/commit/221ce21a209e2e32a3eaaa2d9e28ca81842fad20) on 0.71, though i think that fix comes with more feature-wise changes?

**Reply by kelset** [reply]
let's chat about this during the weekly sync

**Comment by robhogan** [comment]
Since RN 0.71.0 went out with Terser as the default JS minifier for the first time (replacing Uglify-ES), we've had reports of and fixed two issues in source maps generated by Metro+Terser.

The issues amount to source maps being (usually?) invalid in specific entries, and potentially incompatible with any tooling that inspects them - one known issue is that `react-native-bundle-visualizer` stopped working.

I released a Metro hotfix [0.73.8](https://github.com/facebook/metro/releases/tag/v0.73.8) with just those fixes over Metro 0.73.7 (currently distributed with RN 0.71.3). Due to exact version pinning of RNCLI and Metro, we'd need an RN patch release for these fixes to reach RN users.

**Reply by robhogan** [reply]
Opened a PR to the 10.x branch of the CLI: https://github.com/react-native-community/cli/pull/1841

**Reply by robhogan** [reply]
10.2.0 has been released with Metro 0.73.8. Opened a PR to pull this into `0.71-stable`: https://github.com/facebook/react-native/pull/36251

Edit: Now merged.

**Reply by kelset** [reply]
thanks Rob 🙏

**Comment by asherLZR** [comment]
Should we pick https://github.com/facebook/hermes/commit/059d08582de630af4c7b0be9925353296afbd96f? Exists since 0.70.

**Comment by mikeduminy** [comment]
Hey there, there is an issue in Hermes that prevents one from loading large, complex bundles in development. The change is to simply increase the max size of the stack.

The change hasn't been merged yet but it targets the stable branch within Hermes for 0.71. My guess is some coordination needs to happen to first merge this then bump this in RN. I'm happy to help where ever needed.

https://github.com/facebook/hermes/pull/925

**Reply by kelset** [reply]
looks like that commit wouldn't be enough: https://github.com/facebook/hermes/issues/135#issuecomment-1444320351

**Reply by mikeduminy** [reply]
I know it looks that way but this commit is not the same as the commit referenced in that comment. https://github.com/facebook/hermes/commit/03f2dffc1d0ef8b2360a6790ad425ce4013e4de3#diff-6c71d832d338aa663651a4747ab28c7de265ccc722fc2587d2749d18b69aa96cL118 actually removes the `kMaxNumRegisters`, whereas this PR backports the new default to 0.71 (and there's another [PR to 0.70](https://github.com/facebook/hermes/pull/924)).

**Comment by NickGerleman** [comment]
https://github.com/facebook/react-native/commit/c03de97fb44f0aecbec8f930bd99fe26f37f9648 makes FlatList permissve of ArrayLike non-array data, which better matches the behavior of 0.70 when using that (e.g. `Realm.Results`).

**Reply by tido64** [reply]
Maybe a bit late since this has already merged, but shouldn't the type be `readonly ArrayLike<ItemT> | null | undefined` in the `.d.ts` file?

**Reply by NickGerleman** [reply]
`ArrayLike` is already readonly.

**Comment by kelset** [comment]
https://github.com/facebook/react-native/pull/36323/files

**Comment by priyeshshah11** [comment]
Could this be included too? https://github.com/facebook/react-native/pull/36340

**Reply by kelset** [reply]
I tried including it but it triggers a conflict, so we'll skip it for now
